### PR TITLE
Allow to reuse the same cluster

### PR DESCRIPTION
**Pull request by mudler**
This is particularly useful for local iteration, so there is no need to re-pull images all the times

Still wip, teardown of deps ain't that easy as expected

Signed-off-by: Ettore Di Giacinto <edigiacinto@suse.com>

### CODE REVIEW

File: tests/e2e/e2e_suite_test.go [hunk 1]
```diff
@@ -138,3 +145,27 @@ var _ = BeforeSuite(func() {
...
 	})
 })
+
+var _ = AfterSuite(func() {
+
+	if os.Getenv("NO_TEARDOWN") != "" {
+		Skip("No teardown")
+	}
+
+	err := exec.Command("system-tools", "remove", "--force", "-c", kubeconfig).Run()
```
Comment (Itxaka) [inline tests/e2e/e2e_suite_test.go]: this doesnt exists on my system :o

**Comment by Itxaka** [comment]
I really dont think this is very useful. Maybe we should change the e2e-tests script to be e2e-setup which setups the cluster with the pre-reqs instead so you can have a cluster setup with the dependencies? The you can run the e2e tests only or drop the clsuter or manually install the rancheros-operator instead? Kind of setting up a vm with ann the needed stuff then running the tests on that vm or letting the user interact with it?

File: tests/e2e/e2e_suite_test.go (hunk 1 above)
Comment (mudler) [reply tests/e2e/e2e_suite_test.go]: yep, In the long run I plan to move it to a kube pod handling deletion.. just trying it out locally at this point

**Comment by mudler** [comment]
> I really dont think this is very useful.

I do need a way to store the images locally, trashing the cluster each time is not optimal at all for local development, just sharing my setup here

> Maybe we should change the e2e-tests script to be e2e-setup which setups the cluster with the pre-reqs instead so you can have a cluster setup with the dependencies? The you can run the e2e tests only or drop the clsuter or manually install the rancheros-operator instead? Kind of setting up a vm with ann the needed stuff then running the tests on that vm or letting the user interact with it?

good point, maybe I can make it the other way around and make the setup optional instead of the teardown

**Comment by Itxaka** [comment]
> I do need a way to store the images locally, trashing the cluster each time is not optimal at all for local development, just sharing my setup here

mmmh, I didnt think of that, maybe my internet speed is too fast to notice LOL

Isnt some kind of local proxy registry that we can reuse for local deployment so it picks the images from there instead of grabbing them from the internet?


Abit convoluted but seems we kind of can: https://maelvls.dev/docker-proxy-registry-kind/

**Comment by mudler** [comment]
> > I do need a way to store the images locally, trashing the cluster each time is not optimal at all for local development, just sharing my setup here
> 
> mmmh, I didnt think of that, maybe my internet speed is too fast to notice LOL

You wouldn't notice now that we aren't touching much and using tests, but if you need to run them sequentially it eventually becomes cumbersome

> 
> Isnt some kind of local proxy registry that we can reuse for local deployment so it picks the images from there instead of grabbing them from the internet?

kind actually supports to side load image from the host, but that way we have to pull all images from the docker daemon beforeahead and it is too much convoluted imho

anyhow.. need to implement some proper cleanup logic that takes care of the tests resources that gets created during the timeframe so you wouldn't get into false positive by rerunning the test suite twice, will have a look at it later

**Comment by Itxaka** [comment]
can confirm that the proxy methods works and its kind of cool. Seems something that we could implement for local development as its simple enough!

**Review by Itxaka — APPROVED** [review]
nice

**Pull request merged**